lib/gvl-tracing.rb

```diff
@@ -28,6 +28,50 @@
 require_relative "gvl_tracing/version"
 
 require "gvl_tracing_native_extension"
+require "json"
 
 module GvlTracing
+  class << self
+    @@path = "/tmp/gvl-tracing.json"
+
+    def start(file)
+      @@path = file
+      GvlTracingNativeExtension.start(@@path)
+    end
+
+    def stop
+      thread_list = Thread.list
+      GvlTracingNativeExtension.stop
+      set_thread_name(thread_list)
+    end
+
+    private
+
+    def set_thread_name(thread_list)
+      output_file = File.open(@@path)
+      output = output_file.read.split("\n").map do |event|
+        parse_event = JSON.parse(event)
+        thread_id = parse_event.dig("args", "name")
+
+        next(parse_event) unless thread_id
+
+        thread = thread_list.find { |t| t.native_thread_id.to_s == thread_id.strip }
```

Comment:
It might be faster to precompute a hash e.g. `thread_list.index_by { |t| native_thread_id.to_s }`  (minus the ActiveSupport-ism). Otherwise you end up with O(N * M) complexity.